PostfixConfigGenerator.py

```diff
@@ -0,0 +1,417 @@
+    if not smtp_tls_fingerprint_digest:
+      smtp_tls_fingerprint_digest = "md5"
+
+    # Will be set to True if at least one of the provided fingerprints
+    # uses an hash alg that matches the one used by Postfix.
+    digest_matched = False
+
+    for tlsa in lst:
+      # hex_fprint will contain the fingerprint in hex str format: abcd12...
+      hex_fprint = None
+
+      if "hash_alg" in tlsa:
+        # Fingerprint is given in the policy, just need to translate it in the
+        # Postfix format: xx:xx:xx:...
+
+        if tlsa["data_format"] == "hex":
+          hex_fprint = tlsa["data"]
+        elif tlsa["data_format"] == "b64":
+          hex_fprint = bin_to_hexstr(base64.b64decode(tlsa["data"]))
+        else:
+          raise ValueError("Unknown data format for %s's EE fingerprint: %s" %
+                           (domain,tlsa["data_format"]))
+
+        if tlsa["hash_alg"] == smtp_tls_fingerprint_digest:
+          digest_matched = True
+      else:
+        # The policy contains only the full entity (pubkey or leaf certificate)
+        # so the fingerprint must be calculated here.
+
+        if tlsa["data_format"] == "hex":
+          bin_data = hexstr_to_bin(tlsa["data"])
+        elif tlsa["data_format"] == "b64":
+          bin_data = base64.b64decode(tlsa["data"])
+        else:
+          raise ValueError("Unknown data format for %s's EE fingerprint: %s" %
+                           (domain,tlsa["data_format"]))
+
+        if smtp_tls_fingerprint_digest == "sha1":
+          dgst = hashlib.sha1()
+        elif smtp_tls_fingerprint_digest == "sha256":
+          dgst = hashlib.sha256()
+        elif smtp_tls_fingerprint_digest == "sha512":
+          dgst = hashlib.sha512()
+        else:
+          raise ExistingConfigError("Postfix is using an "
+                           "unsupported hash algorithm "
+                           "for smtp_tls_fingerprint_digest ({hash}), the EE "
+                           "{entity}'s fingerprint for {domain}'s TLSA "
+                           "can't be calculated. "
+                           "Fix Postfix configuration or modify the "
+                           "TLS policy.".format(domain=domain,
+                           entity=tlsa["entity"],
+                           hash=smtp_tls_fingerprint_digest))
+
+        dgst.update(bin_data)
+        hex_fprint = dgst.hexdigest()
+        digest_matched = True
+
+      res.append(hexstr_to_hexstr_with_colon(hex_fprint))
+
+    if not digest_matched:
+      raise ExistingConfigError("None of the TLSAs of {domain} matches the "
+                       "current hash algorithm used by Postfix for "
+                       "fingerprints matching: {hash}. "
+                       "Fix Postfix configuration or modify the TLS "
+                       "policy.".format(domain=domain,
+                       hash=smtp_tls_fingerprint_digest))
+
+    return res
+
+  def _build_smtp_tls_policy_maps_line(self,domain,policy):
```

Comment:
I really like the extension to having 'client' fingerprints and dane.  We should probably talk about how is best to incorporate some of this, or to defer till later (when say we have totally working certbot functionality first).